### PR TITLE
Cancel running queries when Spectacles gets interrupted

### LINKED ISSUE #88: Kill running queries on error or KeyboardInterrupt
If a `KeyboardInterrupt` occurs or an error happens while we're running query tasks in the SQL validator, we should kill any running query tasks [via the API](https://docs.looker.com/reference/api-and-integration/api-reference/v3.1/query#kill_running_query).

### PR DESCRIPTION
**Pull request by DylanBaker**
Closes #88

### CODE REVIEW

**Comment by DylanBaker** [comment]
@joshtemple This is a rough initial implementation. As discussed, as far as I can tell, it does cancel the queries as expected, except for when they are removed from the `running_query_tasks` queue in `_get_query_results`.

Maybe that's good enough for a first pass, but I'd obviously much rather get this done properly. It's no longer such as issue now that we have the `--concurrency` flag.

Very open to suggestions on ways to tackle this. I'll obviously also  give it further thought.

File: spectacles/validators.py [hunk 1]
```diff
@@ -244,13 +261,33 @@ def validate(self, mode: str = "batch") -> List[SqlError]:
...
+                    self.client.cancel_query_task(session, query_task_id)
+                )
+                cancel_query_tasks.append(task)
 
-        await session.close()
+            await asyncio.gather(*cancel_query_tasks)
 
-        return errors
+            raise SpectaclesException(
+                "Spectaces was interrupted. "
```
Comment (joshtemple) [inline spectacles/validators.py]: Can we prepend 1-2 newlines here so it doesn't look like this?
```
^CSpectaces was interrupted.```

File: spectacles/cli.py [hunk 2]
```diff
@@ -130,6 +130,11 @@ def wrapper(*args, **kwargs):
                 + "\n"
             )
             sys.exit(error.exit_code)
+        except KeyboardInterrupt as error:
+            logger.debug(error, exc_info=True)
+            logger.info("ctrl-c")
```
Comment (joshtemple) [inline spectacles/cli.py]: Don't think we need this, plus it's not always Ctrl+C.

File: spectacles/cli.py [hunk 3]
```diff
@@ -130,6 +130,11 @@ def wrapper(*args, **kwargs):
...
+            logger.info("Encountered KeyboardInterrupt.")
```
Comment (joshtemple) [inline spectacles/cli.py]: Something a bit more readable like, "Spectacles was manually interrupted."

File: spectacles/validators.py (hunk 1 above)
Comment (joshtemple) [reply spectacles/validators.py]: Also, to match above, can we do "Spectacles was manually interrupted."?

**Pull request merged**